docs/architecture/gas/README.md

```diff
@@ -21,11 +21,49 @@ there to estimate the safe values of the new parameters. This section is for you
 if you are adding new features such as a new pre-compiled method or other host
 functions.
 
+A high-level summary of what types of parameters exist follows right here.
 
-<!-- TODO: ## Action parameters-->
-<!-- TODO: ## WASM parameters-->
-<!-- TODO: ## Non-gas parameters -->
-<!-- TODO: - Gas economics config-->
-<!-- TODO: - Gas economics config-->
-<!-- TODO: - Storage usage config-->
-<!-- TODO: - Smart contract limits-->
\ No newline at end of file
+## Action Costs
+
+Actions are executed in two steps. First, an action is verified and inserted to
+an action receipt, which is sent to the receiver of the action. The `send` fee
+is paid for this. It is charged either in `process_transaction` when the action
+is part of a fresh transaction, or inside
+[logic.rs](https://github.com/near/nearcore/blob/14b8ae2c7465444c9b672a23b044c00be98f6e34/runtime/near-vm-logic/src/logic.rs)
+through `pay_action_base(..)` when the action is generated by a function call.
+
+The second step is action execution. It is charged in `fn apply_action(..)`.
+
+The send fee is meant to cover the cost to validate an action and transmit it
+over the network. It can be different for local receipts (sender = receiver).
+The execution cost has to cover everything required to apply the action to the
+blockchain's state.
+
+In conclusion, each action parameter is split into three costs, `send_sir`,
+`send_not_sir`, and `execution`. They should be estimated, defined, and charged
+separately. The reality is that today almost all actions are estimated as a
+whole and the parameters are split 50/50 between send and execution cost,
+without discrimination on local vs remote receipts.
```

Comment:
does this mean that `send_sir` and `send_not_sir` are always the same?  then we could do something like below maybe:

```suggestion
without discrimination on local vs remote receipts i.e. `send_sir` == `send_not_sir`.
```